projects/angular/src/data/tree-view/models/tree-node.model.ts

```diff
@@ -46,21 +47,38 @@ export abstract class TreeNodeModel<T> {
     this.selected.complete();
   }
 
+  toggleChildrenDisable() {
```

Comment:
The word "toggle" in this method name implies that it flips the disabled state of the children from true to false or vice versa. It does not toggle the disabled state though. It just sets it match this node. I would rename this method to `propagateDisabledToChildren` or similar for clarity.

Reply:
Like this: https://github.com/vmware-clarity/ng-clarity/commit/b081f58c05c90d731d650de9c9c7f83239fad678

Reply:
You could also make the get accessor traverse up the tree (look at parent nodes) to see if there is a disabled ancestor node. I think there is an issue with the current code: You should be able to enable a node without enabling all of its children.

Reply:
Disable functionality should work only towards children nodes. There is no use case where node should disable a parent, but all this is philosophical anyway and is not part of the current requirements.

Reply:
Disabled accessor added in commit 64af64b

Reply:
>There is no use case where node should disable a parent

I did not say that a node should disable a parent. I said a child node could check its parent nodes to see if it (the child node) is disabled.